lib/lease-manager.js

```diff
@@ -33,45 +59,48 @@ class LeaseManager extends EventEmitter {
    */
   initialize() {
     return this.provider.initialize().then((result) => {
-      this.status = STATUS.READY;
       this.data = result.data;
       this.lease_duration = result.lease_duration;
-      this.emit('ready');
-      this.error = null;
+
       this._renew();
+      if (this.status !== STATUS.READY) {
+        this.emit('ready');
```

Comment:
I think the state transition to `READY` needs to happen before the call to `this._renew()`. Otherwise, if the renew timeout fires immediately (the secret has no lease duration) and errors, the data will be cleared.